tests/components/application-connector/internal/testkit/test-api/requestparams.go

```diff
@@ -0,0 +1,49 @@
+package test_api
+
+import (
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+type ExpectedRequestParameters struct {
+	Headers         map[string][]string
+	QueryParameters map[string][]string
+}
+
+func RequestParameters(expectedRequestParams ExpectedRequestParameters) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			for key, expectedVals := range expectedRequestParams.Headers {
+				actualVals := r.Header.Values(key)
+				if !equal(actualVals, expectedVals) {
+					handleError(w, http.StatusBadRequest, "Incorrect additional headers. Expected %s header to contain %v, but found %v", key, expectedVals, actualVals)
+					return
+				}
+			}
+
+			queryParameters := r.URL.Query()
+			for key, expectedVals := range expectedRequestParams.QueryParameters {
+				actualVals := queryParameters[key]
+				if !equal(actualVals, expectedVals) {
+					handleError(w, http.StatusBadRequest, "Incorrect additional query parameters. Expected %s query parameter to contain %v, but found %v", key, expectedVals, actualVals)
+					return
+				}
+			}
+
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func equal(a, b []string) bool {
```

Comment:
Hmm, I remember that the actual headers were a bit longer than the expected ones. For example:
```
r.Header: map[Accept-Encoding:[gzip] Authorization:[Basic dXNlcjpwYXNzd2Q=] Hkey1:[hval1] Hkey2:[hval21 hval22] User-Agent:[Go-http-client/1.1] X-B3-Parentspanid:[bbdb9c9aefff2856] X-B3-Sampled:[0] X-B3-Spanid:[cd0f02f5b0fce301] X-B3-Traceid:[7d0e12550001eb8e8949e5b46f07442e] X-Envoy-Attempt-Count:[1] X-Forwarded-Client-Cert:[By=spiffe://cluster.local/ns/test/sa/default;Hash=16d230dcc9da00aa18a616cabc351f34c34d06474c228704367f38e81c6853a8;Subject=\"\";URI=spiffe://cluster.local/ns/kyma-system/sa/central-application-gateway] X-Forwarded-For:[127.0.0.6] X-Forwarded-Proto:[http] X-Request-Id:[6037182d-f655-4af4-9a31-4c04106b14d0]]

expectedHeaders: map[Hkey1:[hval1] Hkey2:[hkey21 hkey22]]
```

In this case, this test should not pass if we are checking their equity. That's a little suspicious. The only reason it may work is that we disable Istio so Envoy does not add anything there. If so, in my opinion, it's a bit risky to leave the strict equality here